crates/sc-consensus-subspace/src/slot_worker.rs

```diff
@@ -106,6 +107,11 @@ where
         // (default state), it also accounts for DSN sync
         (!self.force_authoring && self.inner.is_major_syncing())
             || self.pause_sync.load(Ordering::Acquire)
+            || self
+                .domain_sync_finished
+                .as_ref()
```

Comment:
The `SeqCst` ordering has an impact on read and write performance, because it imposes a total order across all threads:
https://doc.rust-lang.org/std/sync/atomic/enum.Ordering.html#variant.SeqCst